docs/cross-chain-payments-and-execution.md

```diff
@@ -0,0 +1,77 @@
+## Cross chain payments and execution
+
+This document suggests an approach of using `UserOp`s to perform cross-chain payments and execution. At a very high level this approach can be thought of as a cross-chain state channel, where we have some state, that when fully signed can be submitted to an adjudicator contract to claim the funds based on the outcome of that state.
+
+However instead of implementing the state channel framework ourselves, we're making use of some of ERC 4337 infrastructure. Instead of signing a state, participants sign a **UserOp** that contains the state information. Instead of having a specific "adjudicator" contract, we have the **entrypoint** and the **BridgeWallet SCW** to handle adjudicating funds. Instead of making a on-chain call to challenge on the adjudicator, we can submit the `UserOp` to an entrypoint to trigger a challenge.
+
+Since we can specify the UserOp calldata, we can include whatever additional state information we want in that calldata. That lets us embed payment or execution information for different chains in one UserOp. When submitted to chain the UserOp will behave depending on the chain Id and the embedded payment or execution information.
+
+# Payments
+
+We embed payment information in the UserOp using a `PaymentChainInfo`
+
+```sol
+/// Contains information to execute a payment
+struct PaymentChainInfo {
+  uint chainId;
+  address entrypoint;
+  State paymentState;
+}
+
+```
+
+Whenever a fully signed UserOp containing a `PaymentChainInfo` is submitted to chain, it forces a challenge on that chain. This means that once you have a fully signed UserOp you know you can always use it to get your funds via challenge. In the happy path, we would expect participants to just exchange signed states afterwards, so they can discard the UserOp.
+
+## Cross-chain Payment Example
+
+Let's say we have Alice, Bob,and Irene. Alice and Irene have a BridgeWallet on chain A, Bob and Irene have a BridgeWallet on Chain B. Alice and Irene have both signed a state with a balance of `[A:5,I:5]`, and bob and Irene have likewise have a signed state with a balance of`[B:5,I:5]`. Let's say Alice wants to pay Bob.
+
+1. Alice creates two new unsigned states `[A:4,I:6]` and `[B:6,I:4]`
+2. Alice includes them in a UserOp(wrapped in a `PaymentChainInfo`), and signs the UserOp. She sends this to Bob and Irene.
+3. Bob and Irene validate the UserOp and also sign it, and broadcast it.
```

Comment:
What do you mean by "broadcast it" in this context? Where are Bob and Irene sending the signed `UserOp`?

Reply:
>Are you replacing the hashlock mechanism with signatures, or just adding to it?

In this example we are replacing the hashlock mechanism, since we're making a cross-chain payment that doesn't use HTLCs.